compiler/rustc_lint/src/context/diagnostics.rs

```diff
@@ -187,6 +187,14 @@ pub(super) fn builtin(
         BuiltinLintDiagnostics::UnexpectedCfgName((name, name_span), value) => {
             let possibilities: Vec<Symbol> =
                 sess.parse_sess.check_config.expecteds.keys().copied().collect();
+
+            #[allow(rustc::potential_query_instability)]
```

Comment:
I don't think this instability can be dismissed. It's possible that two or more different cfgs have the same value that we are searching for, in which case we should either suggest both or none.

Reply:
ah, we still need to keep `#[allow(rustc::potential_query_instability)]` to remove the warning because we use `iter()`...

Reply:
This is an unfortunate limitation of the lint, it doesn't detect the following `sort()` call. Can you add a comment explaining that this is fine since we sort the list just below.

Reply:
I'm curious why `keys()` and `values()` don't have `potential_query_instability`, and `into_keys()`, `into_values()` have that property, according to the document, they are all in `arbitrary order`.

Reply:
It's missing the custom rustc attribute `#[rustc_lint_query_instability]` like in the `into_keys` function: https://github.com/rust-lang/rust/blob/635124704849eeead4e3a7bb6e663c5351571d93/library/std/src/collections/hash/map.rs#L484-L486

This could added in another PR.

Reply:
yeah, I know it's missing `#[rustc_lint_query_instability]`, I was wondering if there is any special reason here, but they are actually all iter in the inner.

I will create another pr later.